pymc/gp/cov.py

```diff
@@ -663,6 +690,28 @@ def power_spectral_density(self, omega: TensorLike) -> TensorVariable:
         pow = pt.power(5.0 + pt.dot(pt.square(omega), pt.square(ls)), -1 * D52)
         return (num / den) * pt.prod(ls) * pow
 
+    @staticmethod
+    def required_num_eigenvectors(L: float, ls: float) -> int:
+        r"""Number of eigenvectors in Hilbert space that approximate the GP well.
```

Comment:
Could add some more detail here?  Would be kind of mysterious if you're not using HSGPs, or know how to choose L.  

Another idea (take it or leave it) would be to have this put into a single function in `hsgp_approx.py` that does this all in one go, something like

```python
def required_num_eigenvectors(cov_func, L, ls):
    if cov_func.__class__.__name__ == "ExpQuad":
        return ceil(1.75 * L / ls)
    elif ...:
        ...
```
which maybe is better because the docstring can be in one place and the logic and purpose is isolated in one spot instead of spread out.  I wonder if the recommendations may evolve too, since in the paper they didn't try for higher than 1D inputs.

Reply:
Even more convenient would be to have this function called automatically if `m` is not passed to HSGP (along with a healthy warning message informing users what is going on).

Reply:
and `c` too (but maybe not in this PR)

Reply:
Yeah, I'll do this